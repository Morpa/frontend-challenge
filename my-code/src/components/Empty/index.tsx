import Link from 'next/link'
import * as S from './styles'

const Empty = () => (
  <S.Wrapper>
    <Link href="/movie/[id]" as={'/movie/tt0068646'} passHref>
      <a>
        <img alt="List of movies is empty" src="/img/empty.png" />
        <S.Description>Don’t know what to search?</S.Description>
        <S.Info>Here’s an offer you can’t refuse</S.Info>
      </a>
    </Link>
  </S.Wrapper>
)

export default Empty
