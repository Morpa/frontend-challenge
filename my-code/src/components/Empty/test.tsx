import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Empty from '.'

describe('<Empty />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Empty />)

    expect(
      screen.getByRole('img', { name: /List of movies is empty/i })
    ).toHaveAttribute('src', '/img/empty.png')

    expect(
      screen.getByRole('heading', { name: /Donâ€™t know what to search/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
