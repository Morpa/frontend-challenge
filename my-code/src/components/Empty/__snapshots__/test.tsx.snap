// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  padding: 5.6rem;
}

.c0 img {
  width: 100%;
  height: auto;
  object-fit: cover;
  overflow: hidden;
  border-radius: 0.4rem;
}

.c0 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  font-size: 2.0rem;
  color: #7A8C99;
  margin-top: 1.6rem;
}

.c2 {
  font-size: 2.0rem;
  color: #7A8C99;
  margin-top: 1.6rem;
}

@media (min-width:768px) {
  .c0 img {
    max-width: 30rem;
  }
}

@media (max-width:768px) {
  .c1 {
    font-size: 1.8rem;
  }
}

@media (max-width:768px) {
  .c2 {
    font-size: 1.6rem;
  }
}

<div
  class="c0"
>
  <a
    href="/movie/tt0068646"
  >
    <img
      alt="List of movies is empty"
      src="/img/empty.png"
    />
    <h3
      class="c1"
    >
      Don’t know what to search?
    </h3>
    <p
      class="c2"
    >
      Here’s an offer you can’t refuse
    </p>
  </a>
</div>
`
